### DISCUSSION TITLE
Build instructions?

### DISCUSSION THREAD

**Discussion by nofishonfriday** [FossifyOrg/General-Discussion#134] [Q&A] [answered]
Hi,

I'm thinking about contributing to this project (with code/PRs) but I'm unsure how to build the apps.
To be more exact I'm unclear how to utilise the `Commons` repo when building say the Calendar app via Android studio.

I dont have that much experience with Android development, I'm coming from desktop development (Java/C++ mainly) and from there I'm familiar with the concept of [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) for common modules but here the `Commons` module seems to work differently.

Thanks.

**Comment by Aga-C** [comment, marked as answer]
If you don't need to touch anything in Commons, it's enough to just run the app in Android Studio - Commons will be fetched from Jitpack as an external dependency.

However, if you'd need to do anything in Commons, you can check this commented out code in Gallery: https://github.com/FossifyOrg/Gallery/blob/master/settings.gradle.kts#L24 and do something similar in any app you're working on to utilize local copy of Commons.

**Comment by nofishonfriday** [comment]
Thanks for the reply.

I'm planning to work on the calendar app, thinking about/trying to implement my 'old' [FR](https://github.com/SimpleMobileTools/Simple-Calendar/issues/1911), also it's a good occassion learning more about Android development anyway. :)

From a brief look, it seems I also need to work in Commons so I went with your second method but got a build error, see screenshot.
Any idea how to proceed?

![simple calendar build error](https://github.com/FossifyOrg/General-Discussion/assets/7658194/63f539c7-5751-455e-b438-1ff99f7473a9)

**Reply by Aga-C** [reply]
In `gradle/libs.versions.toml` find `gradlePlugins-agp` and update the version, so both Commons and Calendar will have the same one. Then run Gradle sync once again.

However, please be aware, that due to some changes in the recent Commons version, build will fail. If you don't need the code for which there are errors (probably for reminders you won't need it), you can temporarily comment it out.

**Comment by nofishonfriday** [comment]
At first I had quite a few build errors but after updating Android Studio to latest version and commenting out a few lines which caused errors (as you said) I get it to build with a local copy of `Commons` without error now, yay.

Thanks a lot for your guidance.

**Reply by nofishonfriday** [reply]
PS: Maybe consider adding these build guidelines to the readme.md or in the Wiki for other interested devs. :)

**Comment by nofishonfriday** [comment]
@Aga-C After some break from coding (and setting up a new computer etc.) I'm revisiting this and found the code you [linked above](https://github.com/orgs/FossifyOrg/discussions/134#discussioncomment-8634112) seems to have changed.
So I'd like to ask: what's the current procedure to build with a local copy of Commons (rather than fetching from Jitpack)?

**Reply by Aga-C** [reply]
Check this: https://github.com/FossifyOrg/Gallery/commit/760d227b0bc0873e1d33770177a1b91c70b867e5. It has been removed from the app's code, but you can still use this method.

**Comment by naveensingh** [comment]
Subscribe to https://github.com/FossifyOrg/General-Discussion/issues/285. Will make it easier to include Commons locally.

**Reply by nofishonfriday** [reply]
Thanks for the quick replies.
And sorry for the noob questions. :)
I got it to build from the local Commons copy now (I think) after commenting out a few lines which caused build errors, but shouldn't I now see the Commons module source code in the Calendar project? Otherwise how would I debug/set break points in Commons?

<img width="1573" height="986" alt="studio64_2Q4sRKrbPs" src="https://github.com/user-attachments/assets/8f7a69f5-8ce1-4a59-bb22-422b03fab151" />

**Reply by naveensingh** [reply]
_Now_ you only need to add `includeBuild("../Commons")` in your `settings.gradle.kts` file for side-by-side development. The manual dependency substitution is not needed anymore. Commons should appear as a module in the same window once you run **File ➜ Sync Project with Gradle Files**. If it doesn't, something else is broken.

Also, for your specific use case (develop and debug side-by-side), https://github.com/FossifyOrg/General-Discussion/issues/285 won't be much help. That issue is about publishing changes in commons lib to the local cache, which will then get picked by all other Fossify projects without any changes to their build configuration.

Later, I'll add these instructions to the commons README and app build instructions in all other repos.

**Reply by naveensingh** [reply]
If you get any errors related to divider binding, comment out those lines and then build (needs fixing, I haven't worked on it yet).

**Reply by nofishonfriday** [reply]
Ah I was missing the File ➜ Sync Project with Gradle Files step.
Thanks.
